packages/tangle/timemanager.go

```diff
@@ -82,6 +93,12 @@ func (t *TimeManager) Shutdown() {
 		t.tangle.Events.Error.Trigger(errors.Errorf("failed to persists LastConfirmedMessage (%v): %w", err, cerrors.ErrFatal))
 		return
 	}
+
+	t.shutdownOnce.Do(func() {
+		t.mu.Lock()
```

Comment:
Is there any reason for this lock? Shouldn't the `Once` assure that the function is not called twice at the same time?